main.go

```diff
@@ -102,10 +117,76 @@ If the passed version does not exist, an error is signaled.`)
 		`A comma separated list of code migration to perform:
 - "autoalias": Apply auto aliasing to the provider.`)
 
+	cmd.PersistentFlags().StringVar(&context.UpstreamProviderName, "upstream-provider-name", "",
+		"The name of the upstream provider.")
+
 	return cmd
 }
 
 func main() {
 	err := cmd().Execute()
 	contract.IgnoreError(err)
 }
+
+// Adapted from https://github.com/carolynvs/stingoftheviper/blob/main/main.go
+func initializeConfig(cmd *cobra.Command) error {
+	v := viper.New()
+
+	// Set the base name of the config file, without the file extension.
+	v.SetConfigName(defaultConfigFilename)
+
+	// Set as many paths as you like where viper should look for the
+	// config file. We are only looking in the current working directory.
+	v.AddConfigPath(".")
+
+	// Attempt to read the config file, gracefully ignoring errors
+	// caused by a config file not being found. Return an error
+	// if we cannot parse the config file.
+	if err := v.ReadInConfig(); err != nil {
+		// It's okay if there isn't a config file
+		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			return err
+		}
+	}
+
+	// When we bind flags to environment variables expect that the
+	// environment variables are prefixed, e.g. a flag like --number
+	// binds to an environment variable STING_NUMBER. This helps
+	// avoid conflicts.
+	v.SetEnvPrefix(envPrefix)
+
+	// Environment variables can't have dashes in them, so bind them to their equivalent
+	// keys with underscores, e.g. --upstream-provider-name to UPGRADE_UPSTREAM_PROVIDER_NAME
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+
+	// Bind to environment variables
+	// Works great for simple config names, but needs help for names
+	// like --favorite-color which we fix in the bindFlags function
+	v.AutomaticEnv()
+
+	// Bind the current command's flags to viper
+	bindFlags(cmd, v)
+
+	return nil
+}
+
+// Bind each cobra flag to its associated viper configuration (config file and environment variable)
+func bindFlags(cmd *cobra.Command, v *viper.Viper) {
+	cmd.Flags().VisitAll(func(f *pflag.Flag) {
+		// Determine the naming convention of the flags when represented in the config file
+		configName := f.Name
+		// If using camelCase in the config file, replace hyphens with a camelCased string.
```

Comment:
should this read:
```suggestion
		// If using hyphenated names in the config file, replace hyphens with a camelCased string.
```
?

Reply:
Honestly I think it's easier to be strict and and require with the config value format to match the flag, i.e. (`upstream-provider-name` is valid, `upstream_provider_name` or `upstreamProviderName` are not`). I'll change it to that for now